### ISSUE TITLE
Network connection was lost

### ISSUE THREAD

**Issue by fvvliet** [socketio/socket.io-client-swift#403]
Hello Socket.IO users,

We use Socket.IO-client-swift in our Swift (XCode 7.3.1) project and tried to update from v4.1.6 to v6.1.3  After updating we receive **Network connection was lost** errors as soon as we call socket.connect(). We cannot find what's causing this so had to go back again to v4.1.6. Any ideas what might be the issue ?

Thanks for any advice
Frank

**Comment by nuclearace** [comment]
Can you turn on logging? I've not seen this before.

**Comment by CoderEic** [comment]
2016-06-23 11:46:22.937 WebSocketTest[1596:519919] LOG SocketIOClient: Adding handler for event: connect
2016-06-23 11:46:22.940 WebSocketTest[1596:519919] LOG SocketIOClient: Adding handler for event: currentAmount
2016-06-23 11:46:22.940 WebSocketTest[1596:519919] LOG SocketIOClient: Adding engine
2016-06-23 11:46:22.945 WebSocketTest[1596:519919] LOG SocketEngine: Starting engine. Server: ws://115.29.193.48:8088
2016-06-23 11:46:22.945 WebSocketTest[1596:519919] LOG SocketEngine: Handshaking
2016-06-23 11:46:22.949 WebSocketTest[1596:519961] LOG SocketEnginePolling: Doing polling request
2016-06-23 11:46:23.311 WebSocketTest[1596:519959] ERROR SocketEnginePolling: The network connection was lost.
2016-06-23 11:46:23.312 WebSocketTest[1596:519959] ERROR SocketEngine: The network connection was lost.
2016-06-23 11:46:23.313 WebSocketTest[1596:519959] ERROR SocketIOClient: The network connection was lost.
2016-06-23 11:46:23.315 WebSocketTest[1596:519959] LOG SocketIOClient: Handling event: error with data: (
    "The network connection was lost."
)

**Comment by fvvliet** [comment]
We have the same log results

**Comment by nuclearace** [comment]
@fvvliet It looks like @CoderEic is trying to connect to a plain websocket server, which is expected to fail since this library requires a socket.io server. So I suspect your problem may be different.

**Comment by fvvliet** [comment]
[2016-06-28 07:44:12.337] AppDelegate.logProjectSettings():356 CHAT: https://api.twypapp.com/test/chat
2016-06-28 07:44:19.491 Testing[39187:1496610] LOG SocketIOClient: Adding handler for event: connect
2016-06-28 07:44:19.491 Testing[39187:1496610] LOG SocketIOClient: Adding handler for event: disconnect
2016-06-28 07:44:19.491 Testing[39187:1496610] LOG SocketIOClient: Adding handler for event: reconnect
2016-06-28 07:44:19.491 Testing[39187:1496610] LOG SocketIOClient: Adding handler for event: chat_message
2016-06-28 07:44:19.491 Testing[39187:1496610] LOG SocketIOClient: Adding handler for event: business_event
2016-06-28 07:44:19.491 Testing[39187:1496610] LOG SocketIOClient: Adding engine
2016-06-28 07:44:19.492 Testing[39187:1496610] LOG SocketEngine: Starting engine
2016-06-28 07:44:19.492 Testing[39187:1496610] LOG SocketEngine: Handshaking
2016-06-28 07:44:19.492 Testing[39187:1496610] LOG SocketEnginePolling: Doing polling request
2016-06-28 07:44:20.237 Testing[39187:1496803] ERROR SocketEnginePolling: The network connection was lost.
2016-06-28 07:44:20.237 Testing[39187:1496803] ERROR SocketEngine: The network connection was lost.
2016-06-28 07:44:20.238 Testing[39187:1496803] ERROR SocketIOClient: The network connection was lost.
2016-06-28 07:44:20.238 Testing[39187:1496803] LOG SocketIOClient: Handling event: error with data: (
    "The network connection was lost."
)
2016-06-28 07:44:20.238 Testing[39187:1496803] LOG SocketEngine: Engine is being closed.
2016-06-28 07:44:20.238 Testing[39187:1496803] LOG SocketEnginePolling: Created POST string: 1:1
Received event: error, with items: Optional(<_TtCs21_SwiftDeferredNSArray 0x7fc65a9010d0>(
The network connection was lost.
)
)
2016-06-28 07:44:20.239 Testing[39187:1496803] LOG SocketEnginePolling: Doing polling request
2016-06-28 07:44:20.447 Testing[39187:1496786] LOG SocketIOClient: Starting reconnect
2016-06-28 07:44:20.447 Testing[39187:1496786] LOG SocketIOClient: Handling event: reconnect with data: (
    "The network connection was lost."
)

**Comment by nuclearace** [comment]
@fvvliet How are you creating the socket? Are you using the `.Path` option?

**Comment by fvvliet** [comment]
Hello @nuclearace , see below

        let debugging = true
        
        let connectionParams  = [
            "reconnectAttempts" : -1, // infinite retries
            "reconnectWait" : 5,
            "reconnects" : true,
            "log" : debugging
        ]
        
        let url = NSURL(string: Configuration.chatURL)
        self.socket = SocketIOClient(socketURL: url!, options: connectionParams)

**Comment by cprovatas** [comment]
Same issue here but only when connected to wifi, not using cellular.  Any help would be appreciated

**Comment by Komal2905** [comment]
Same issue here too

**Comment by kinza88** [comment]
hello @fvvliet 
Did you find any solution to this?

**Comment by fvvliet** [comment]
Nope, we sticked to the working older version. And a year ago the project stopped so no update on this issue...

**Comment by kinza88** [comment]
okay.. thankyou.

**Comment by butterpuf** [comment]
@fvvliet Can you tell the older version that does not have this issue?

**Comment by fvvliet** [comment]
@butterpuf :  The project is more than a year old and the app has been retrieved from the store but what I found in the code is this. Hope it helps...
pod 'Socket.IO-Client-Swift', :git => 'https://github.com/bosmagit/socket.io-client-swift.git', :branch =>'v5.2.3_swift2.3'

**Comment by butterpuf** [comment]
ok thankyou

**Comment by AkAyush** [comment]
Hi Socket team,

I am also facing the same issue , can you guys please suggest a solution.
